component/file_cache/file_cache.go

```diff
@@ -920,6 +961,14 @@ func (fc *FileCache) ReadInBuffer(options internal.ReadInBufferOptions) (int, er
 	// The file should already be in the cache since CreateFile/OpenFile was called before and a shared lock was acquired.
 	// log.Debug("FileCache::ReadInBuffer : Reading %v bytes from %s", len(options.Data), options.Handle.Path)
 
+	options.Handle.Lock()
+	var err error
+	options.Handle, err = fc.DownloadFile(internal.DownloadFileOptions{Name: options.Handle.Path, Handle: options.Handle})
```

Comment:
Now that all you need to pass in to `DownloadFile` is the handle, maybe you can drop `DownloadFileOptions` and just pass the handle on its own?